gcp/modules/gcp-stackdriver-monitoring/main.tf

```diff
@@ -27,7 +28,7 @@ resource "template_dir" "resources" {
     project_id                = "${var.project_id}"
     domain_name               = "${var.domain_name}"
     ssl_enabled_uptime_checks = "${var.ssl_enabled_uptime_checks}"
-    notification_email        = "${var.notification_email != "" ? var.notification_email : var.auth_user_email}"
+    notification_email        = "${(var.env == "dev" && var.auth_user_email != "") ? var.auth_user_email : var.notification_email}"
```

Comment:
I would prefer to avoid changing behavior based on `env` as it seems confusing.

I think the desired precedence is:

1. `notification_email`, if defined -- use `alerts+[stg|prd]@` no matter what for `stg|prd`
2. `auth_user_email`, if defined -- use `tyler@` directly for `dev-tyler`
3. a new variable, `fallback_email`?  -- when no env-specific `notification_email` and no user-specific `auth_user_email` is available

The new variable could default to `alerts+dev@` or maybe `alerts+fallback@`?

Another approach might be to manually set `auth_user_email` in `.gitlab-ci.yml` for environments like `gcp-dev` and `dev-doe`. CI is the special case, so let it do the special thing.

Reply:
@mrtyler Replied in thread.
@stepanstipl Let's leave poor code as it is for now, I thought we all agreed to replace it with Terraform resources as soon as all Stackdriver primitives are supported.

Reply:
ok fair enough :)